google/stackdriver_monitoring/spectator_client.py

```diff
@@ -151,12 +160,67 @@ def __init__(self, options):
     self.__host = options['host']
     self.__prototype = None
     self.__default_scan_params = {'tagNameRegex': '.+'}
+    self.__previous_scan_lock = threading.Lock()
+    self.__previous_scan = {} if options.get('log_metric_diff') else None
```

Comment:
Does `options.get('log_metric_diff')` throw a `KeyError` if `'log_metric_diff'` isn't present? Make it default to `None` if it does.